### PR TITLE
Dapp schema overhaul

### PR DESCRIPTION
**Pull request by jordajm**
This PR breaks out each schema category into its own schema with a version.

There's also a lightweight adapter that maps listing data that was created with a deprecated schema to a valid schema so the listing can be displayed, edited, etc.

NOTE: This PR impacts a lot of important stuff, so we should test it thoroughly. 

I have tested creating/editing listings (including fractional usage listings), purchasing those listings, viewing purchases listings, etc., but each one of these had some nuance to it and could use another test, plus whatever scenarios/workflows that I forgot about.

Notes about the schemas:
- each has a `dappSchemaId` (which needs to be the same as the file name), a `listingType`, and a `category`
- fractional listings have a `slotLength` and a `slotLengthUnit` that are editable and have reasonable defaults based on the schema type

See designs for [desktop](https://app.zeplin.io/project/59fa2311bac7acbc8d953da9/screen/5c05b70961f917792a6d40c6) and [mobile](https://app.zeplin.io/project/59fa2311bac7acbc8d953da9/screen/5c05b7bf51f8e40181490cda)

### Mobile Screen Shots:

screenshot 2018-12-10 00 17 18

------------

screenshot 2018-12-10 00 17 33

-----------

### Desktop Screen Shots:

screenshot 2018-12-10 00 24 03

----------

screenshot 2018-12-10 00 24 27

### CODE REVIEW

**Comment by jordajm** [comment]
@franckc @micahalcorn @wanderingstan - here's a PR with some ideas we can discuss at our 1:30pm PST call tomorrow...

**Comment by jordajm** [comment]
Closing in favor of https://github.com/OriginProtocol/origin/pull/1026
All the important stuffs from this PR will be implemented there.

**Comment by jordajm** [comment]
Oops - closed the wrong PR... reopening.

**Comment by jordajm** [comment]
@micahalcorn @franckc @wanderingstan - this PR is a 🐻. I'd be happy to walk through it with any of you if that would help the review process. 

Basically, the first 115-ish files are the new/updated/removed schemas, then you'll see the dapp changes (mostly listing-create and listing-detail components), then a couple of origin-js changes to accommodate the new fractional usage schema stuff.

**Comment by micahalcorn** [comment]
I’ll test the DApp and defer to @franckc or @wanderingstan on the code review.

**Comment by jordajm** [comment]
☝️ thanks @micahalcorn - good find ☝️

File: origin-dapp/src/components/listing-create.js [hunk 1]
```diff
@@ -545,19 +635,85 @@ class ListingCreate extends Component {
...
+                  />
+                </h2>
+                <button
+                  onClick={() => this.setState({
+                    step: this.STEP.PICK_CATEGORY,
+                    selectedSchemaId: null
+                  })}
+                  className="mobile-back-btn"
+                >
+                  <span>&#10094;</span>
```
[the remark this answers is not in the document]
Comment (jordajm) [reply origin-dapp/src/components/listing-create.js]: Sorry @micahalcorn 😅  - weird local styling issue made this look fine in the mobile emulator - fixed now

File: origin-dapp/src/components/listing-create.js [hunk 2]
```diff
@@ -47,42 +47,36 @@ class ListingCreate extends Component {
...
-      DETAILS: 2,
-      AVAILABILITY: 3,
-      BOOST: 4,
-      PREVIEW: 5,
-      METAMASK: 6,
-      PROCESSING: 7,
-      SUCCESS: 8,
-      ERROR: 9
+      PICK_CATEGORY: 1,
+      PICK_SCHEMA: 2, // NOTE: this is a mobile-only step
```
[the remark this answers is not in the document]
Comment (jordajm) [reply origin-dapp/src/components/listing-create.js]: Agreed - changed step name to `PICK_SUBCATEGORY`

**Comment by micahalcorn** [comment]
@jordajm your SVGs are now available in Zeplin. Add those, if you don't mind, and then check out Franck's comments before we merge. 🙏

**Comment by franckc** [comment]
LGTM - Yes, I think this is ready to be merged ! @micahalcorn did you want to do further testing or should we go ahead and merge this in ?

**Pull request merged**